src/GameplaySystem.cpp

```diff
@@ -11,11 +11,50 @@ GameplaySystem::GameplaySystem() {
 
 void GameplaySystem::update() {
     // Update game state here
+    EventSystem::queue_event (
+        Event (
+            EventType::IDLE_GAME_OBJECT_EVENT,
+            "object_id", 1,
+            "rotation_rad", 0.01f
+        )
+    );
 }
 
 void GameplaySystem::handle_load(const Event& e) {
     // Initialize gameplay here
     std::cout << "Gameplay initialized here" << std::endl;
+
+    // Ship 1
+    EventSystem::queue_event(
+        Event(
+            EventType::ADD_GAME_OBJECT_EVENT,
```

Comment:
I think these can be a little less abstract, but more specific. Otherwise `GameplaySystem` wont' know what objects are what. So I think this event being called `ADD_EXAMPLE_SHIP` would be better. And this way, we don't need to pass information like vertex and fragment shader paths, which only the rendering system needs to know.